CONTRIBUTING.md

```diff
@@ -76,10 +76,14 @@ such as links to decisions about the structure of the code and notes features th
 The use of comments to note problems or future fixes is discouraged, and merges containing them will not be approved.
 Instead, create an issue to track the required change.
 
-<!--
 ## Testing
-This section is to be drafted as part of #11 "Add testing harness"
--->
+
+Testing an application framework which primarily deals with network interactions is always a difficult problem.
+The design of mewbot using a number of components that communicate indirectly through dataclasses partially
+helps with this, allowing us to craft inputs and verify outputs.
+
+Tests are performed using the `pytest` testing framework, with tests located in the `tests` folder.
+Merge requests should not reduce the overall test coverage of the framework, as calculated by Sonar
```

Comment:
SHOULD or MUST? Presumably exceptions must be agreed in the PR?

Reply:
I actually don't know how Sonar configures the gate, so I didn't want to write something stricter than what the tests say